src/window.rs

```diff
@@ -1,11 +1,12 @@
 use crate::drm::DrmDisplay;
 use drm::control::{framebuffer::Handle as FramebufferHandle, Device as ControlDevice};
-use gbm::{BufferObjectFlags, Format as BufferFormat, Surface};
+use gbm::{BufferObject, BufferObjectFlags, Format as BufferFormat, Surface};
 
 pub struct Window {
     pub(crate) gbm_surface: Surface<FramebufferHandle>,
     pub(crate) drm_display: DrmDisplay,
-    pub(crate) crtc_set: bool,
+    pub(crate) frame_count: usize,
+    previous_bo: Option<BufferObject<FramebufferHandle>>,
```

Comment:
I have a slight preference to spell out the name if it's not terribly long
```suggestion
    previous_buffer_object: Option<BufferObject<FramebufferHandle>>,
```

or

```suggestion
    previous_buffer: Option<BufferObject<FramebufferHandle>>,
```